### PR TITLE
[MM-48317] Fix: Upgrade to enterprise when settling delinquency from Professional

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
The Purchase Modal will always try to subscribe you to the product that's next in line - ie, if you are on Starter, it tries to subscribe you to Professional. If you're on Professional, it will try to subscribe you to Enterprise. 

This causes issues when someone on Professional tries to settle their delinquent account, as once their cc is entered, they are upgraded to Enterprise. 

This PR makes it so that if the subscription is delinquent (and has been for fewer than 90 days) it will not attempt to upgrade, as that subscription hasn't been downgraded yet. If it's been delinquent greater than 90 days, or is not delinquent at all, the existing behaviour remains.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48317

#### Related Pull Requests

- Has server changes (please link here)
- Has mobile changes (please link here)

#### Release Note

```release-note
None
```

### CODE REVIEW

**Review by neallred — COMMENTED** [review]
Code LGTM, but why do we need to upgrade if delinquent longer than 90 days?

**Comment by neallred** [comment]
Nvm, if I understand correctly, its because after 90 days they get downgraded to freemium.

File: components/purchase_modal/purchase_modal.tsx [hunk 1]
```diff
@@ -609,6 +607,16 @@ class PurchaseModal extends React.PureComponent<Props, State> {
...
+    selectedProduct = () => {
+        // If the subscription is not past 90 days delinquent, we don't need to upgrade them as they're still on the correct plan
+        if (this.props.isDelinquencyModal && !this.props.isCloudDelinquencyGreaterThan90Days) {
+            return findProductInDictionary(this.props.products, this.props.productId);
+        }
+
+        // If not delinquent, or they've been downgraded (ie, passed 90 days), the modal functions as normal
+        return this.state.selectedProduct;
+    }
+
```
Comment (marianunez) [inline components/purchase_modal/purchase_modal.tsx:619]: This works but we could also add that condition where the `selectedProduct` is initially set [here](https://github.com/mattermost/mattermost-webapp/blob/8f81a8278f3d3eddd4ff087794a5525888d591e0/components/purchase_modal/purchase_modal.tsx#L166-L174 ) This way we leave all the logic of selecting the current product in the same place and don't have issues of in the future accessing `this.state.selectedProduct` incorrectly

File: package-lock.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
   "requires": true,
   "packages": {
     "": {
+      "name": "mattermost-webapp",
```
Comment (M-ZubairAhmed) [inline package-lock.json:8]: can we revert this change for the scope of this pr

**Review by neallred — APPROVED** [review]
:shipit:

**Review by marianunez — APPROVED** [review]
LGTM, thanks @nickmisasi!
